runner/logger.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+)
+
+type LogMessage struct {
+	Level   string `json:"level"`
+	Message string `json:"message"`
+	Type    string `json:"type"`
+}
+
+func (m LogMessage) toJsonString() string {
+	jsonStruct, err := json.Marshal(m)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(jsonStruct)
+}
+
+type CronRunnerLogger struct {
+	Format      string
+	Destination string
+	Type        string
+	logger      *log.Logger
+}
+
+func NewCronRunnerLogger(format, dest string) *CronRunnerLogger {
+	logOpts := 0
+
+	var logWriter logWriter
+	var logger *log.Logger
+
+	if dest == "os.Stdout" {
+		if format == "json" {
+			logWriter = NewLogWriter("cron-control", "cron-control-runner", false)
+		} else {
+			// original log format
+			logWriter = NewLogWriter("cron-control", "cron-control-runner", true)
+		}
+		logger = log.New(logWriter, "", logOpts)
+	} else {
+		fileWriter := NewFileWriter(dest, "cron-control", "cron-control-runner")
+
+		logger = log.New(fileWriter, "", logOpts)
+	}
+
+	return &CronRunnerLogger{
+		format,
+		dest,
+		"cron-control-runner",
+		logger,
+	}
+}
+
+func (crl *CronRunnerLogger) Printf(format string, v ...interface{}) {
+	if crl.Format == "json" {
+		msgStr := fmt.Sprintf(format, v...)
+
+		// Should be using level specific functions to set the `Level` (i.e.: `Debugf()`) but we're
+		// doing it this way fit with how `logger` is used in `runner.go`
```

Comment:
Should we just change how `logger` is called? Or is it not worth it / too much effort?

Reply:
I think there's definitely value in making more "correct" but as it is now, there is only one way it is being used (as in all logs are basically `DEBUG`) so I wasn't sure if it was worth doing it now or waiting to till the use case expands a little. I don't really know the planned direction for this runner daemon